temporalio/client.py

```diff
@@ -207,6 +211,26 @@ def data_converter(self) -> temporalio.converter.DataConverter:
         """Data converter used by this client."""
         return self._config["data_converter"]
 
+    @property
+    def rpc_metadata(self) -> Mapping[str, str]:
+        """Headers for every call made by this client.
+
+        Do not use mutate this mapping. Rather, set this property with an
+        entirely new mapping to change the headers.
+        """
+        return self.service_client.config.rpc_metadata
+
+    @rpc_metadata.setter
+    def rpc_metadata(self, value: Mapping[str, str]) -> None:
+        """Update the headers for this client.
+
+        Do not mutate this mapping after set. Rather, set an entirely new
```

Comment:
I feel like an explicit setter method is better than this setter + comment to not mutate.